### ISSUE TITLE
Multidimensional input: adding phonological features

### ISSUE THREAD

**Issue by abylouw** [TensorSpeech/TensorFlowTTS#675]
Hi,

I am trying an experiment where instead of just adding a segment id (character/phoneme) as input, I would like to add some phonological features as well. I am looking at using FastSpeech2 as a base, but I am struggling to find a simple way to change the architecture to add multidimensional input.

Has anyone played around with such an architecture and willing to share some tips?

Thanks

**Comment by abylouw** [comment]
I currently have about 56 binary phonological features. My plan now is to make the vocab size = 56 and then input a matrix of these features for each phoneme. For each phonone I then mask the phonological features that are off, gather the others from the embedding and sum them all to get a phonological feature vector of size 256 for each phoneme.

Any thoughts on this scheme?

**Comment by dathudeptrai** [comment]
@abylouw that scheme is ok i think, WDYT @ZDisket

**Comment by ZDisket** [comment]
@abylouw @dathudeptrai Looks good to me.

**Comment by abylouw** [comment]
Hi @dathudeptrai,

Is the attention mask at https://github.com/TensorSpeech/TensorFlowTTS/blob/cd3a5e1f9915fa7dd646771fd50fe6fef94fe9fc/tensorflow_tts/models/fastspeech2.py#L153
to mask out padding? or what is it for?

Thank you

**Comment by dathudeptrai** [comment]
@abylouw yes :D
